apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java

```diff
@@ -112,176 +107,43 @@ private static Coverage transformPartA(MetricRegistry metricRegistry, Beneficiar
 
     coverage.addContract(TransformerUtilsV2.referenceCoverage("contract1", MedicareSegment.PART_A));
 
-    // coverage
-    // .getClass()
-    // .setSubGroup(TransformerConstants.COVERAGE_PLAN)
-    // deh end
-    // .setSubPlan(TransformerConstants.COVERAGE_PLAN_PART_A);
+    beneficiary.getMedicareBeneficiaryId().ifPresent(value -> coverage.setSubscriberId(value));
 
-    Optional<String> mbiUnhashedCurrent = beneficiary.getMedicareBeneficiaryId();
-    if (mbiUnhashedCurrent.isPresent()) coverage.setSubscriberId(mbiUnhashedCurrent.get());
+    setTypeAndIssuer(coverage);
 
-    coverage.setType(
-        new CodeableConcept()
-            .addCoding(
-                new Coding()
-                    .setCode("SUBSIDIZ")
-                    .setSystem("http://terminology.hl7.org/CodeSystem/v3-ActCode")));
+    setCoverageRelationship(coverage, SubscriberPolicyRelationship.SELF);
 
-    coverage
-        .addPayor()
-        .setIdentifier(new Identifier().setValue(TransformerConstants.COVERAGE_ISSUER));
+    createCoverageClass(
+        coverage, CoverageClass.SUBGROUP, TransformerConstants.COVERAGE_PLAN, Optional.empty());
 
-    coverage.setRelationship(
-        new CodeableConcept()
-            .addCoding(
-                new Coding()
-                    .setCode("self")
-                    .setSystem("http://terminology.hl7.org/CodeSystem/subscriber-relationship")
-                    .setDisplay("Self")));
+    createCoverageClass(
+        coverage,
+        CoverageClass.SUBPLAN,
```

Comment:
What is the reason for coding these as "subplan" and "subgroup" over "plan" and "group".  The only reason I ask is that CARIN has slices defined for `plan` and `group` but nothing else.

Reply:
I saw the methods invoked (.setSubPlan() & .setSubGroup()) and assume that the Coverage. should be the same; wrong assumption on my part. Fixed.